src/Sylius/Behat/Context/Setup/ProductReviewContext.php

```diff
@@ -39,32 +41,136 @@
+     */
+    public function thisProductHasAReviewTitledAndRatedAddedByCustomerWhichIsNotAcceptedYet(
+        ProductInterface $product,
+        $title,
+        $rating,
+        CustomerInterface $customer
+    ) {
+        $review = $this->createProductReview($product, $title, $rating, $title, $customer, null);
+
+        $this->productReviewRepository->add($review);
+    }
+
+    /**
+     * @Given /^(this product)(?:| also) has accepted reviews rated (\d+), (\d+), (\d+), (\d+) and (\d+)$/
+     * @Given /^(this product)(?:| also) has accepted reviews rated (\d+), (\d+) and (\d+)$/
+     */
+    public function thisProductHasAcceptedReviewsRated(ProductInterface $product, ...$rates)
+    {
+        $customer = $this->sharedStorage->get('customer');
+        foreach ($rates as $key => $rate) {
+            $review = $this->createProductReview($product, 'Title '.$key, $rate, 'Comment '.$key, $customer);
+            $this->productReviewRepository->add($review);
+        }
+    }
+
+    /**
+     * @Given /^(this product)(?:| also) has review rated (\d+) which is not accepted yet$/
+     */
+    public function itAlsoHasReviewRatedWhichIsNotAcceptedYet(ProductInterface $product, $rate)
+    {
+        $customer = $this->sharedStorage->get('customer');
+        $review = $this->createProductReview($product, 'Title', $rate, 'Comment', $customer, null);
+        $this->productReviewRepository->add($review);
+    }
+
+    /**
+     * @Given /^(this product) also has review rated (\d+) which is rejected$/
+     */
+    public function itAlsoHasReviewRatedWhichIsRejected(ProductInterface $product, $rate)
+    {
+        $customer = $this->sharedStorage->get('customer');
+        $review = $this->createProductReview($product, 'Title', $rate, 'Comment', $customer, 'reject');
+        $this->productReviewRepository->add($review);
+    }
+
+    /**
+     * @param ProductInterface $product
+     * @param string $title
+     * @param int $rating
+     * @param string $comment
+     * @param CustomerInterface|null $customer
+     * @param string $transition
+     *
+     * @return ReviewInterface
+     */
+    private function createProductReview(
+        ProductInterface $product,
+        $title,
+        $rating,
+        $comment,
+        CustomerInterface $customer = null,
+        $transition = 'accept'
+    ) {
+        /** @var ReviewInterface $review */
         $review = $this->productReviewFactory->createNew();
-        $review->setTitle('title');
-        $review->setRating(5);
+        $review->setTitle($title);
+        $review->setRating($rating);
+        $review->setComment($comment);
         $review->setReviewSubject($product);
+        $review->setAuthor($customer);
 
         $product->addReview($review);
 
-        $this->productReviewRepository->add($review);
+        if (null !== $transition) {
+            $stateMachine = $this->stateMachineFactory->get($review, 'sylius_product_review');
```

Comment:
Shouldn't we have `final class ProductReviewTransitions` with constants as we have for all other state machine graphs?